File: Design_Pattern_and_Principles/AdapterPatternExample/Adaptees.java

class PayPalGateway {
    public void sendPayment(double amountInUSD) {
        System.out.println("Processing payment of $" + amountInUSD + " through PayPal.");
    }
}

class StripeGateway {
    public void chargeInCents(int amountInCents) {
        System.out.println("Charging total of " + amountInCents + " cents through Stripe.");
    }
}

File: Design_Pattern_and_Principles/AdapterPatternExample/Adaptees.cs
using System;

class PayPalGateway {
    public void SendPayment(double amountInUSD) {
        Console.WriteLine("Processing payment of $" + amountInUSD + " through PayPal.");
    }
}

class StripeGateway {
    public void ChargeInCents(int amountInCents) {
        Console.WriteLine("Charging total of " + amountInCents + " cents through Stripe.");
    }
}